tsconfig.json

```diff
@@ -2,11 +2,16 @@
   "compilerOptions": {
     "target": "es6",
     "module": "commonjs",
+    "moduleResolution": "node",
```

Comment:
Same here, please make linting changes in a separate PR.

Reply:
ok. Do you know why the tests are failing?
I added sourceMap flags to the build commands to try and get some better error messages.

From testing on my own repo with `ACTIONS_STEP_DEBUG` secret set to true, it seems to be working correctly
when the token has the correct permissions.

I think what's happening is the token used in the tests doesn't have enough permissions to list release drafts
or to create new releases.

Reply:
On a separate note, I don't think this action should create the release if it doesn't exist.

There are already several simple and very complex actions that create releases:
- https://github.com/actions/create-release
- https://github.com/release-drafter/release-drafter
- https://github.com/marketplace/actions/auto-release-draft

Creating releases you usually want some sort of changelog computed from the commit messages,
or issues or pull requests. The above actions can do that. In our case the body is a static string at best.

To make things worse the release we create is automatically published as well (it is not a release draft).

I think the scope of this action should be limited to simply uploading assets to existing releases or release drafts.
Leave release creation to other tasks that more designed for that purpose.

Reply:
It seems the tests were failing because of a missing await.
It was returning a Promise instead of the result of the promise.